core/storage/auth/gae_models.py

```diff
@@ -0,0 +1,93 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Models for managing user authentication."""
+
+from __future__ import absolute_import  # pylint: disable=import-only-modules
+from __future__ import unicode_literals  # pylint: disable=import-only-modules
+
+from core.platform import models
+
+base_models, = models.Registry.import_models([models.NAMES.base_model])
+datastore_services = models.Registry.import_datastore_services()
+
+
+class UserIdByFirebaseAuthIdModel(base_models.BaseModel):
+    """Stores the relationship between user ID and Firebase auth ID.
+
+    Instances of this class are keyed by Firebase auth ID.
+    """
+
+    user_id = datastore_services.StringProperty(required=True, indexed=True)
+
+    @staticmethod
+    def get_deletion_policy():
+        """Model has data to delete corresponding to users: id and user_id."""
+        return base_models.DELETION_POLICY.DELETE_AT_END
+
+    @staticmethod
+    def get_model_association_to_user():
+        """Currently, the model holds IDs relevant only for backend that should
+        not be exported.
+        """
+        return base_models.MODEL_ASSOCIATION_TO_USER.NOT_CORRESPONDING_TO_USER
+
+    @classmethod
+    def get_export_policy(cls):
+        """Model doesn't contain any data directly corresponding to a user.
+        Currently, the model holds authentication details relevant only for
+        backend, and no exportable user data. It may contain user data in the
+        future.
```

Comment:
The association itself may be useful--will we be ensuring that we export which Firebase account is associated with a user's Oppia account later on?

Reply:
I don't think there's a need, users cannot interact with Firebase directly. If they want to make changes to their account (unlink/delete/etc), they would have to use an API we provide them.

@vojtechjelinek what do you think?

Reply:
I'm not sure either, I guess if they cannot use the ID for any purpose I would not export it. 

From what I understand Firebase is de facto another place where we save our user data (similar to NDB Datastore now or Firestore in the future) this fact should be mentioned in our ToS but I don't think we need to provide all the details to the users.

Reply:
Got it, so Firebase can be thought of as a secondary blackbox database owned by Oppia. The database is updated by using the Firebase SDK. The only data stored in the database is details necessary for identifying the user, and the only data Oppia depends on is this single "Auth ID" value.


This makes me think that our options are pretty straight-forward:
-   Wipeout: Delete Firebase account through the SDK and models associated to it.
-   Takeout: Nothing, Firebase does not hold any profile information, just the bare necessities for identifying them.

Does that sound OK?

Reply:
I think from the user's perspective, it seems like Oppia is connected to their phone number/Gmail account (and that's how we think about it from an authentication perspective, as well). It's that association that I think has value to export as part of takeout. While the association itself may not contain sensitive user data, it's still quite relevant to a user's understanding of how their data is being used which is why this seems relevant.

I'm also fine with this being a non-blocking change for the Firebase integration, but I think we should be including it in the export. Can you add a line item to your document task list and/or file an issue to track exporting this association? If it's still questionable to you, I'm happy to discuss it further when I return in the new year.

Your thinking for wipeout matches mine. :)

Reply:
Hmm just for clarification, are you suggesting we takeout all the user-identifying information when a Takeout is requested? Or would the Auth ID be enough?

Reply:
I think that we should export all the user-identifying information if that would be too hard and the user can use their Auth ID to retrieve their information exporting just Auth ID might be enough.

Reply:
Got it, I've talked with @vojtechjelinek and he's OK with adding this as a separate PR that would block Firebase authentication launch. This has been tracked in #11462 as well.